snow/consensus/snowball/binary_snowflake.go

```diff
@@ -7,17 +7,12 @@ import "fmt"
 
 var _ Binary = (*binarySnowflake)(nil)
 
-func newBinarySnowflake(alphaPreference, alphaConfidence, beta, choice int) binarySnowflake {
+func newBinarySnowflake(alphaPreference int, terminationConditions []terminationCondition, choice int) binarySnowflake {
```

Comment:
It seems that we aren't testing the cases where:
1. We call `RecordPoll` with `count < alphaPreference`
2. We call `RecordPoll` with `choice != preference` and `count < alphaPreference`

Reply:
Specifically:
1.
```diff
 	if count < sf.alphaPreference {
-		sf.RecordUnsuccessfulPoll()
 		return
 	}
```
2. 
```diff
- 	if count < sf.alphaPreference {
-		sf.RecordUnsuccessfulPoll()
-		return
- 	}
```

Reply:
https://github.com/zimmski/go-mutesting 👀 

fwiw the draft as is only adds testing for unary snowflake, will add this for the rest of the cases and make sure to mutation test the package